docs/modules/record-encryption/proc-delete-lost-records.adoc

```diff
@@ -0,0 +1,54 @@
+// file included in the following:
+//
+// record-encryption-guide/con-lost-kek.adoc
+
+[id='proc-delete-lost-records-{context}']
+= Deleting Lost Records
+
+[role="_abstract"]
+This section describes how to delete records from a Kafka topic that are unrecoverable owing to a lost KEK.
+
+WARNING: Delete records from a topic is an irreversible option.
+
+.Prerequisites
+
+* Name of the topic affected.
+* Failing offset (this is reported in the error message reported by the client).
+* Admin access to the Kafka Cluster with `READ`, `DESCRIBE` and `DELETE` permission for the affected topic.
+* Access to the Kafka CLI utilities `kafka-console-consumer.sh`, kafka-get-offsets.sh` and `kafka-delete-records.sh`.
+
+.Procedure
+
+. Shutdown any applications that are producing to or consuming from the affected topic.
+. Use `kafka-get-offsets.sh` to find the `latest` offset for each partition of the affected topic:
++
+[source,shell]
+----
+kafka-get-offsets.sh --bootstrap-server <bootstrap-server> --topic <topic-name> --time latest
+----
+. Use ``kafka-get-offsets` to find the `latest` offset for each partition of the affected topic:
++
+. Use a binary search technique to discover the offset of the first record that that decrypts successfully beyond the failing offset.
++
```

Comment:
Offsets aren't global. The offset reported above is valid with the partition.  We'd need a time based offset to be able to apply it to all partitions within the topic.